tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersBsonAttributes.cs

```diff
@@ -0,0 +1,72 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using MongoDB.Analyzer.Tests.Common.DataModel;
+using MongoDB.Driver;
+
+namespace MongoDB.Analyzer.Tests.Common.TestCases.Builders
+{
+    public sealed class BuildersBsonAttributes : TestCasesBase
+    {
+        [BuildersMQL("{ \"Cost\" : { \"$exists\" : true } }")]
+        [BuildersMQL("{ \"_id\" : 0 }")]
+        [BuildersMQL("{ \"Weight\" : 1 }")]
+        [BuildersMQL("{ \"Color\" : \"Red\" }")]
+        [BuildersMQL("{ \"$set\" : { \"Volume\" : 22.0 } }")]
+        [BuildersMQL("{ \"$set\" : { \"Width\" : 5 } }")]
+        [BuildersMQL("{ \"DictionaryField\" : { \"$exists\" : true } }")]
+        [BuildersMQL("{ \"Name\" : 1 }")]
+        [BuildersMQL("{ \"ExpiryDate\" : { \"$exists\" : true } }")]
+        [BuildersMQL("{ \"TimeSpanField\" : { \"$exists\" : false } }")]
+        [BuildersMQL("{ \"AppleType\" : { \"$exists\" : false } }")]
+        [BuildersMQL("{ \"GreenAppleCost\" : { \"$exists\" : false } }")]
+        public void Basic_bson_attributes()
+        {
+            _ = Builders<Fruit>.Filter.Exists(f => f.TotalCost, true);
+            _ = Builders<Fruit>.Projection.Exclude(f => f.Name);
+
+            _ = Builders<Fruit>.Sort.Ascending(f => f.Weight);
+            _ = Builders<Fruit>.Filter.Eq(f => f.Color, "Red");
+
+            _ = Builders<Fruit>.Update.Set(f => f.Volume, 22);
+            _ = Builders<Fruit>.Update.Set(f => f.Width, 5);
+
+            _ = Builders<Fruit>.Filter.Exists(f => f.DictionaryField, true);
+            _ = Builders<Flower>.Projection.Include("Name");
+
+            _ = Builders<Fruit>.Filter.Exists(f => f.ExpiryDate, true);
+            _ = Builders<Fruit>.Filter.Exists(f => f.TimeSpanField, false);
+
+            _ = Builders<Apple>.Filter.Exists(a => a.AppleType, false);
+            _ = Builders<GreenApple>.Filter.Exists(g => g.GreenAppleCost, false);
+        }
+
+        [BuildersMQL("{ \"RedAppleCost\" : { \"$exists\" : false } }")]
+        public void Custom_bson_serializer()
+        {
+            _ = Builders<RedApple>.Filter.Exists(g => g.RedAppleCost, false);
+        }
+
+        [NotSupportedBuilders("Unable to determine the serialization information for f => f.Quantity.", version: "(,2.19.0)")]
+        [NotSupportedBuilders("Expression not supported: f.Quantity.", version: "2.19.0")]
+        [NotSupportedBuilders("Unable to determine the serialization information for a => a._AppleID.", version: "(,2.19.0)")]
+        [NotSupportedBuilders("Expression not supported: a._AppleID.", version: "2.19.0")]
+        public void Warnings_due_to_bson_ignore()
+        {
+            _ = Builders<Fruit>.Update.Set(f => f.Quantity, 22);
+            _ = Builders<Apple>.Update.Set(a => a._AppleID, "Apple ID");
+        }
```

Comment:
Also need to add tests for attributes that receive types, for supported and unsupported (should be ignored by analyzer) types.
for example `[BsonId(IdGenerator = typeof(MyIdGenerator))]`

Reply:
Added test cases and the error message are reproduced by analyzer as expected